### DISCUSSION TITLE
Progress bar for watch uploads

### DISCUSSION THREAD

**Discussion by james0r** [uicrooks/shopify-theme-lab#54] [Ideas]
I don't see this option in the themekit commands, so maybe this isn't possible.

I just noticed that I added bootstrap-vue and vue-awesome-swiper globally and already my bundle.js size was 4MB.

The process hung for a bit and then returned with

```
07:34:36 [dev] processing assets/bundle.js
[dev] (assets/bundle.js) request failed after 5 retries with error: Put https://james0r-theme-lab.myshopify.com/admin/api/unstable/themes/120693260478/assets.json: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
```

Eventually I just quit the process and deployed the theme instead and the processed completed.

Then I figured back up the dev server and all was okay. 

Would be nice to get some better output here so we know why it's failing if possible.

**Comment by sergejcodes** [comment]
I looked into the themekit source code and it seems they don't use the progress bar for the `watch` task on purpose.

I'm pretty sure your problem has something to do with the `timeout` setting, which is indicated by `Client.Timeout exceeded while awaiting headers`. By default, it's set to 30s. You can find more in the [Shopify themekit docs](https://shopify.dev/tools/theme-kit/configuration-reference#global-parameters)

You can try a fix it by adding for example `timeout: 100s` to your `shopify.dev.yml`:

```yml
dev:
  ...
  timeout: 100s
```

**Reply by james0r** [reply]
I figured out it was due to the huge bundle size from importing BootstrapVue. That timeout will be helpful tho. Thank you.
